### PR TITLE
http4s extractor

### LINKED ISSUE #8: Http4s Extractor
Http4s is a critical functional component and often people match on a path segment that is a UUID and would like to directly extract. We should provide such an extractor.

### PR DESCRIPTION
**Pull request by JesusMtnez**
Add `FUUIDVar` to extract `UUID` as `FUUID`. It is defined in its own module `fuuid-http4s`.

Close #8

### CODE REVIEW

File: modules/http4s/src/test/scala/io/chrisdavenport/fuuid/http4s/FUUIDVarSpec.scala [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import org.specs2.ScalaCheck
+import org.specs2.mutable.Specification
+
+class FUUIDVarSpec extends Specification with ScalaCheck with FUUIDArbitraries {
+
+  "FUUID Extractor in Path" should {
+
+    "work properly given a valid UUID" in prop { validFuuid: FUUID =>
+
+      (Path(s"/v1/${validFuuid.toString}") match {
```
Comment (ChristopherDavenport) [inline modules/http4s/src/test/scala/io/chrisdavenport/fuuid/http4s/FUUIDVarSpec.scala]: Could we use `show` interpolator here for best practice demonstration.

File: build.sbt [hunk 2]
```diff
@@ -1,25 +1,42 @@
...
+  )
   .enablePlugins(MicrositesPlugin)
   .enablePlugins(TutPlugin)
-  .dependsOn(core)
+  .dependsOn(core, http4s)
 
 val catsV = "1.1.0"
 val catsEffectV = "0.10.1"
 val specs2V = "4.3.2"
+val http4sV = "0.18.5"
```
Comment (ChristopherDavenport) [inline build.sbt]: Please update the http4s version to `0.18.15`

File: modules/docs/src/main/tut/index.md [hunk 3]
```diff
@@ -52,3 +52,28 @@ val laterGreaterThanEarlier : IO[Boolean] = for {
...
+import io.chrisdavenport.fuuid.FUUID
+import io.chrisdavenport.fuuid.http4s.FUUIDVar
+import org.http4s._, org.http4s.dsl.io._
+
+def getEntityByUuid(id: FUUID): IO[String] = ???
+
+val service: HttpService[IO] = 
+  HttpService[IO] {
+    case GET -> Root / "uuid" / FUUIDVar(id) =>
+      Ok(getEntityByUuid(id))
```
Comment (ChristopherDavenport) [inline modules/docs/src/main/tut/index.md]: This is not considerred best practices to return an `IO` inside the `Ok`. could we `flatMap` here?

File: modules/http4s/src/test/scala/io/chrisdavenport/fuuid/http4s/FUUIDVarSpec.scala (hunk 1 above)
Comment (JesusMtnez) [reply modules/http4s/src/test/scala/io/chrisdavenport/fuuid/http4s/FUUIDVarSpec.scala]: :+1: but #25 is required to make it work

File: build.sbt (hunk 2 above)
Comment (JesusMtnez) [reply build.sbt]: My bad, the version was a typo, I write `5` instead of `15`.

**Pull request merged**